src/Flex.php

```diff
@@ -126,6 +127,15 @@ class_exists(__NAMESPACE__.str_replace('/', '\\', substr($file, \strlen(__DIR__)
         $this->config = $composer->getConfig();
         $this->options = $this->initOptions();
 
+        // if Flex is being upgraded, the original operations from the original Flex
+        // instance are stored in the static property, so we can reuse them now.
+        if (property_exists(self::class, 'storedOperations')) {
```

Comment:
to be build on what @fabpot suggested:
```suggestion
        if (property_exists(self::class, 'storedOperations') && self::$storedOperations) {
```